src/runtime_src/core/pcie/driver/linux/include/xocl_ioctl.h

```diff
@@ -520,10 +528,11 @@ enum drm_xocl_ctx_code {
 struct drm_xocl_ctx {
 	enum drm_xocl_ctx_code op;
 	xuid_t   xclbin_id;
+	char	 cu_name[64];
```

Comment:
This is a little awkward... sometimes, cu_index is suffice, sometimes you need cu_name as well to uniquely identify a CU. The code becomes less intuitive and hard to maintain.

@stsoe @saifuddin-xilinx , the reason we need the cu_name is because that, in a multi-slot case, <uuid, cu_index> may refer to a wrong CU since xclbin may be swapped, right? I'm wondering if this is the time to introduce explicit xclbin-lockdown operation (v.s. implicitly locking down by opening the first ctx)? So, user space code will:
1. lock down xclbin in one particular slot
2. find out cu index through sysfs node
3. still talks to driver using cu index
4. unlock xclbin when everything is done

Of course, driver shall check and make sure xclbin is locked down before servicing any ctx request

Or, we separate data structure for ctx allocation and free. In allocation, we always require xclbin uuid and cu_name. In free, we always need the handle back (no cu_index and xclbin uuid any more).

Let me know what you think?

Reply:
Hi Max, 
Actually for legacy context we need xclbin_id and cu_index for a context. That is already present in this data structure. 
Now for the new hw_context we need slot_handle(slot id) and cu_name for a context. Instead of creating a new data structure I have used existing and added cu_name  filed. 

As we are using exchanging slot id between user space and kernel space, I believe we don't need to use sysfs entry to get cu_index. Using slot_id and cu_name driver can maintain this.